### PR TITLE
Add commit message guidelines

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**

### CODE REVIEW

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. `docker pull docker-delivery.stakater.com:443/stakater/developer-handbook:SNAPSHOT-PR-46-1`

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. `docker pull docker-delivery.stakater.com:443/stakater/developer-handbook:SNAPSHOT-PR-46-2`

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. `docker pull docker-delivery.stakater.com:443/stakater/developer-handbook:SNAPSHOT-PR-46-3`

File: git/commit-message-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
+# Commit Message Guideline
+
+A guide to understanding the importance of commit messages and how to write them well.
+
+It may help you to learn what a commit is, why it is important to write good messages, best practices and some tips to plan and (re)write a good commit history.
+
+## What is a "commit"?
+
+In simple terms, a commit is a _snapshot_ of your local files, written in your local repository.
+Contrary to what some people think, [git doesn't store only the difference between the files, it stores a full version of all files](https://git-scm.com/book/eo/v1/Ekkomenci-Git-Basics#Snapshots,-Not-Differences).
```
Comment (kahootali) [inline git/commit-message-guidelines.md:10]: This link is not working

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. `docker pull docker-delivery.stakater.com:443/stakater/developer-handbook:SNAPSHOT-PR-46-4`

**Review by kahootali — APPROVED** [review]

**Pull request merged**